app/main/authorisation.py

```diff
@@ -1,57 +1,90 @@
-from flask import abort, current_app, g
+from abc import ABC, abstractmethod
 
 
-def check_authorised() -> tuple[tuple[str], dict[str]]:
-    """Checks that the user is authorized to submit.
+class AuthBase(ABC):
+    """Auth class ABC. Classes that inherit must implement a constructor, organisations and auth_dict methods."""
 
-    Returns any LAs, places, and fund types that the user is authorized to submit for, otherwise aborts and redirects
-    to 401 (unauthorised) page.
+    @abstractmethod
+    def __init__(self, *args):
+        pass
 
-    :return: the LAs as a tuple, and a dictionary with both the place_names and fund_types
-    """
-    local_authorities, place_names, fund_types = get_local_authority_and_place_names_and_fund_types(g.user.email)
-    if local_authorities is None or place_names is None or fund_types is None:
-        current_app.logger.error(
-            f"User: {g.user.email} has not been assigned any local authorities and/or places and/or fund types"
-        )
-        abort(401)  # unauthorized
-    current_app.logger.info(
-        f"User: {g.user.email} from {', '.join(local_authorities)} is authorised for places: {', '.join(place_names)}"
-        f"and fund types: {', '.join(fund_types)}"
-    )
-    return local_authorities, {"Place Names": place_names, "Fund Types": fund_types}
-
-
-def get_local_authority_and_place_names_and_fund_types(
-    user_email: str,
-) -> tuple[tuple[str] | None, tuple[str] | None, tuple[str] | None]:
+    @abstractmethod
+    def organisations(self) -> tuple[str, ...]:
+        """Return organisations associated with this level of authorisation."""
+        pass
+
+    @abstractmethod
+    def auth_dict(self) -> dict:
+        """Return other details associated with this authorisation."""
+        pass
```

Comment:
I feel like these function names should contain a verb, like `get_organisations()` and `get_auth_dict()` also """Return other details associated with this authorisation.""" seems vauge but prehaps that can't be avoided